src/cairoUtilFuncGen/memory/memoryRead.ts

```diff
@@ -101,9 +112,20 @@ export class MemoryReadGen extends StringIndexedFuncGen {
   }
 
   private getOrCreate(typeToRead: CairoType): GeneratedFunctionInfo {
+    const functionsCalled: FunctionDefinition[] = [this.requireImport(...DICT_READ)];
+
+    // register require imports
+    if (typeToRead instanceof CairoContractAddress) {
+      createImport(...OPTION_TRAIT, this.sourceUnit, this.ast, [], [], {
+        isTrait: true,
+      });
+      createImport(...CONTRACT_ADDRESS, this.sourceUnit, this.ast);
+      functionsCalled.push(createImport(...CONTRACT_ADDRESS_FROM_FELT, this.sourceUnit, this.ast));
+    }
+
     const funcName = `WM${this.generatedFunctionsDef.size}_READ_${typeToRead.typeName}`;
     const resultCairoType = typeToRead.toString();
-    const [reads, pack] = serialiseReads(typeToRead, readFelt, readFelt);
+    const [reads, pack] = serialiseReads(typeToRead, readFelt, readFelt, readAddress, uNread);
```

Comment:
Can you use an approach similar to this one

https://github.com/NethermindEth/warp/blob/2a911fa0f38c15287f233030344228f939801832/src/cairoUtilFuncGen/serialisation.ts#L28-L31

to avoid the pass of extra arguments?

Reply:
also for `storageRead.ts` file

Reply:
But for memory and storage there would be different line of snippet.